src/xrpld/app/tx/detail/InvariantCheck.cpp

```diff
@@ -1120,4 +1123,79 @@ ValidMPTIssuance::finalize(
         mptokensCreated_ == 0 && mptokensDeleted_ == 0;
 }
 
+//------------------------------------------------------------------------------
+
+void
+ValidPermissionedDomain::visitEntry(
+    bool,
+    std::shared_ptr<SLE const> const& before,
+    std::shared_ptr<SLE const> const& after)
+{
+    if (after->getType() != ltPERMISSIONED_DOMAIN)
+        return;
+    auto const& credentials = after->getFieldArray(sfAcceptedCredentials);
+    credentialsSize_ = credentials.size();
+    auto const sorted = credentials::makeSorted(credentials);
+    isUnique_ = !sorted.empty();
+
+    // If array have duplicates then all the other checks are invalid
+    isSorted_ = false;
+
+    if (isUnique_)
+    {
+        unsigned i = 0;
+        for (auto const& cred : sorted)
```

Comment:
Why not just use a normal for loop with `i` here? Easier to ensure that all the indices are kept straight.

Reply:
using `.size()` of general container in the loop is bad idea as many of them non const.  Adding `size` variable or something will lead to the same code.